public/src/reducers/loginReducer.js

```diff
@@ -4,14 +4,19 @@ const defaultAuthState = {
 };
 
 const auth = (state = defaultAuthState, action) => {
-  console.log('Entering loginReducer with state: ', state);
   switch (action.type) {
     case 'USER_LOGIN':
       console.log('Case: USER_LOGIN, Payload: ', action.payload);
       return Object.assign({}, state, {
         isLoggedIn: action.payload.isLoggedIn,
         user: action.payload.user
       });
+    case 'USER_LOGOUT':
+      console.log('Case: USER_LOGOUT, Payload: ', action.payload);
+      return Object.assign({}, state, {
+        isLoggedIn: action.payload.isLoggedIn,
```

Comment:
While the current implementation will work fine, it's over-complicated.  Since we have two action types, `USER_LOGIN` and `USER_LOGOUT`, we don't really need to pass the value of this boolean down using `action.payload.isLoggedIn`.  I suggest either removing this payload information and hard-coding the boolean values into the reducer, or paring down to one action type called something like e.g. `TOGGLE_LOGIN` that can be passed either true or false in its payload.